src/rebar_core.erl

```diff
@@ -84,7 +84,7 @@ process_command(State, Command) ->
                     %% to have both $REBAR_PROFILE set and use 'as' in a command
                     case rebar_state:current_profiles(State) of
                         [default] ->
-                            do([{default, hd(TargetProviders)} | tl(TargetProviders)], State);
+                            do([hd(TargetProviders) | tl(TargetProviders)], State);
```

Comment:
If you're gonna do that, just use `TargetProviders`. I thought this was needed for `as` to be fetched as a command in the `undefined` namespace though!